aptos-move/framework/aptos-stdlib/sources/big_vector.move

```diff
@@ -0,0 +1,360 @@
+module aptos_std::big_vector {
+    use std::error;
+    use std::vector;
+    use aptos_std::table_with_length::{Self, TableWithLength};
+
+    /// Vector index is out of bounds
+    const EINDEX_OUT_OF_BOUNDS: u64 = 1;
+    /// Vector is full
+    const EOUT_OF_CAPACITY: u64 = 2;
+    /// Cannot destroy a non-empty vector
+    const EVECTOR_NOT_EMPTY: u64 = 3;
+
+    /// A scalable vector implementation based on tables where elements are grouped into buckets.
+    /// Each bucket has a capacity of `bucket_size` elements.
+    struct BigVector<T> has store {
```

Comment:
reason to expose this to anything other than the smart vec?

Reply:
I feel we should give ppl option to fine-grained tuning if you are aware of what they're doing.

Reply:
The SmartVec configured with first_bucket_size=0 is equivalent to this, no? I'd prefer not to confuse the devs with one more data structure..

Reply:
K probably exposing only one BigVector but with different cconstructors:

```
module SmartVec {
    pub fun new_auto_config<V>() {}
    pub fun new_with_config(embedded_bucket_size, additional_bucket_size) {}
}
```

Reply:
Let me see how I could make this work to get rid of big vector.